internal/controller/driver_controller.go

```diff
@@ -434,6 +434,14 @@ func (r *driverReconcile) reconcileControllerPluginDeployment() error {
 			utils.LeaderElectionRetryPeriodContainerArg(leaderElectionSpec.RetryPeriod),
 		}
 
+		// csiaddons does not require utils.LeaderElectionContainerArg as its enabled by default and no option to disable it
+		csiAddonsLeaderElection := []string{
+			utils.LeaderElectionNamespaceContainerArg(r.driver.Namespace),
+			utils.LeaderElectionLeaseDurationContainerArg(leaderElectionSpec.LeaseDuration),
+			utils.LeaderElectionRenewDeadlineContainerArg(leaderElectionSpec.RenewDeadline),
+			utils.LeaderElectionRetryPeriodContainerArg(leaderElectionSpec.RetryPeriod),
+		}
```

Comment:
This seems like a duplication of the `leaderElectionArg` setting. What is the difference?
Also, if there is a difference I am missing, please embed these settings in the csiAddons container arg instead of having it here as it only applies to that container.

Reply:
There was a comment added what we have different from the leaderElectionArg, i think you missed that

Reply:
I did, thank you